src/compiler/binder.ts

```diff
@@ -2805,10 +2805,11 @@ namespace ts {
                 return symbol;
             });
             if (symbol) {
-                const flags = isClassExpression(node.right) ?
-                    SymbolFlags.Property | SymbolFlags.ExportValue | SymbolFlags.Class :
-                    SymbolFlags.Property | SymbolFlags.ExportValue;
-                declareSymbol(symbol.exports!, symbol, node.left, flags, SymbolFlags.None);
+                const flags = isIdentifier(node.right) ? SymbolFlags.Alias

```

Comment:
While `isIdentifier` is very narrow and works, we should probably reuse the `isAliasableExpression` condition we use for `exportAssignmentIsAlias` which is used for `bindExportAssignment`, so the rules for what gets an alias vs not is consistent for all possibly-an-alias expressions.

Reply:
Note that `isAliasableExpression` is `isEntityNameExpression || isClassExpression`; previously `module.x = class { }` was not handled as aan alias, but doing so is more consistent. I'm going to see whether it breaks any of our tests.

Reply:
Updated tests, it was an improvement overall. (But it does not, on its own, make `module.exports.x = NS.C` work; I'm looking to see what else needs to happen)